### PR TITLE
Data only models

### PR DESCRIPTION
**Pull request by jthpotz**

### CODE REVIEW

File: src/models/room.ts [hunk 1]
```diff
@@ -1,217 +1,73 @@
...
   }
-
-  /**
-   * Gets the active conversations in this room
-   * @returns {Conversationp[]}
-   */
-  public getConversations(): Conversation[] {
-    return Conversation.getByIDs(Array.from(this.conversationIDs));
+   /**
+    * Check if it's possible to move from this room to target room.
```
Comment (mitchkm) [inline src/models/room.ts:57]: This description is probably outdated, fits more to describe a controller function. The function itself is good though

File: src/models/agent.ts [hunk 2]
```diff
@@ -1,92 +1,93 @@
...
+  }
+  _faction: number;
+  get faction(): Faction {
+    return this.db.retrieveModel(this._faction, Faction) as Faction;
+  }
+  set faction(newFaction: Faction) {
+    this._faction = newFaction ? newFaction.id : -1;
+  }
+  get factionRank(): number {
+    if (this._faction !== -1) {
```
Comment (mitchkm) [inline src/models/agent.ts:81]: I don't think this safely checks all invalid values of _faction

Comment (mitchkm) [reply src/models/agent.ts:81]: the protocol might be to attempt to get the model via iDB and null/undefined check it

File: src/models/agent.ts [hunk 3]
```diff
@@ -97,436 +98,46 @@ export class Agent extends IDObject {
...
-      if (agentInfo.equals(targetInfo)) return agentInfo;
-    }
-    return undefined;
-  }
-
-  /**
-   * Checks if player has specific item.
-   */
-  public hasItem(item: Item): boolean {
+  hasItem(item: Item): boolean {
```
Comment (mitchkm) [inline src/models/agent.ts:139]: following this, add "hasInfo" and "hasGold" functions

File: src/models/information.ts [hunk 4]
```diff
@@ -88,1297 +89,1305 @@ export interface TILQ {
   loc: Room;
   quantity: number;
 }
-
-export class Info extends IDObject {
```
Comment (scm2217) [inline src/models/information.ts:92]: What are the plans for reworking the information model? Disabling the whole with will break a lot of things in events/controller

Comment (mitchkm) [reply src/models/information.ts:92]: I have the information refactor assigned to myself

**Pull request merged**